packages/block-library/src/image/edit.js

```diff
@@ -145,6 +151,25 @@ class ImageEdit extends Component {
 				captionFocused: false,
 			} );
 		}
+
+		if ( url && imageData && ! fileWidth ) {
+			// Old post or just uploaded image. Attempt to update the image props.
+			const sizeFull = get( imageData, [ 'media_details', 'sizes', 'full' ] );
+			const sizeLarge = get( imageData, [ 'media_details', 'sizes', 'large' ] );
+
+			if ( sizeFull && url === sizeFull.source_url ) {
+				if ( sizeLarge && this.imageMatchesRatio( sizeFull.width, sizeFull.height, sizeLarge.width, sizeLarge.height ) ) {
+					// If the full size image was used, and there's a large size that matches the ratio, replace full with large size.
+					this.updateImageURL( sizeLarge.source_url, pick( sizeLarge, [ 'width', 'height' ] ) );
+				} else {
+					// Add image file dimensions.
+					this.props.setAttributes( {
+						fileWidth: get( sizeFull, [ 'actual_size', 'width' ] ) || sizeFull.width,
+						fileHeight: get( sizeFull, [ 'actual_size', 'height' ] ) || sizeFull.height,
```

Comment:
same note here as in the one below about `get()` and its default value…

```js
get( sizeFull, 'actual_size.width', sizeFull.width )
```

Reply:
Yes, preferably to test for "falsy" value from `get()`.